### ISSUE TITLE
Godot for windows exe icon won't change ?

### ISSUE THREAD

**Issue by DebugGameStudios** [godotengine/godot#35735]
Hi there I followed the documentation on the godot's website to change icon's for exe. it don't seem to work.

The window's icon and taskbar icon do change but not the actual executable ? I try godot 3.1 with  Rcedit v1.1.0 and Rcedit v1.1.1 still no joy.... I then try Resource Hacker and same thing happens.

I then also moved to the latest Godot 3.2 and try all above and still not working ?

Im using the stable Releases from Godot website. 
![Report](https://user-images.githubusercontent.com/56065423/73444772-317e6380-4351-11ea-9fa1-c6fdbd266d61.png)

**Comment by Chaosus** [comment]
On Windows, you should download a program like rcedit (https://github.com/electron/rcedit/releases), and add a path to that program in editor settings:
![image](https://user-images.githubusercontent.com/3036176/73445876-c2166d00-436c-11ea-92ba-831af19991b5.png), only after this procedure you would be able to change the exported icon.

**Comment by Chaosus** [comment]
Sorry, I didn't see that you tried to use rcedit.. and closed it automatically. Still you should try it again, it sometimes doesn't work.

**Comment by pouleyKetchoupp** [comment]
I've had similar issues with changing the icon on windows. It might be due to a cache that makes the system still use the old icon whatever you do. Sometimes after a reboot (or several ones) it randomly switches to the right icon...

But I'm pretty sure a fresh export to a different folder would use the correct icon on the exe. Can you try that?

**Comment by DebugGameStudios** [comment]
@Chaosus i been trying it a few times and still nothing lol.

@pouleyKetchoupp I already try the reboots yesterday and today and no joy there. Just tried putting it into a different folder and still nothing lol.

Thank you to both of you for replying :)

**Comment by Chaosus** [comment]
Did you correctly set up the rcedit to Godot?:

![image](https://user-images.githubusercontent.com/3036176/73450392-56d19880-4376-11ea-9272-7d5388471b16.png)

**Comment by DebugGameStudios** [comment]
@Chaosus its set up the same and still having no luck :/ even tried other version of rcedit.

**Comment by KoBeWi** [comment]
I have an ancient version of rcedit that works for me (0.2.0 it says).
Could you try it?
[rcedit-x64.zip](https://github.com/godotengine/godot/files/4134437/rcedit-x64.zip)

**Comment by DebugGameStudios** [comment]
@KoBeWi Still nothing i even tried rcedit v0.1.0

Are the Godot icons hard coded in ? no matter what rcedit version i use it don't seem to change it :/
As in the picture i posted that's the same results im getting.

When i use Resource Hacker it shows all the Godot Sizes here.

Its like the icons are being overridden ???
![Screenshot (35)](https://user-images.githubusercontent.com/56065423/73464045-30abf880-4376-11ea-998c-da2b14c28153.png)

**Comment by DebugGameStudios** [comment]
I also tried using the Resource Hacker to change the Icons and i just get same results like in the first picture i posted in the thread.

**Comment by KoBeWi** [comment]
Wait, are you using an .ico file with all sizes included? I think you need to provide a complete one.

**Comment by Calinou** [comment]
See [ICO file requirements](https://docs.godotengine.org/en/latest/getting_started/workflow/export/changing_application_icon_for_windows.html#ico-file-requirements) in the documentation.

**Comment by DebugGameStudios** [comment]
@KoBeWi  I should of said this from the start i already tried this. and yes im using .ico lol
@Calinou Already done this step lol before i came on here lol i use ImageMagick on a Daily basis with Linux 👍  

Thanks for everyone helping me out... Just to make it clear everything in the Godot documentation for creating a windows icon i have tried before coming onto this thread. 

I also tried rebooting computer and  deleting cache files.

Reinstalling Godot 3.2 with import templates and using all different Rcedit versions

Even went back to Godot 3.1 import templates and using all different Rcedit versions.

Still getting same results lol

I don't wanna give up because its bugging me. and i would like to know what it is lol

**Comment by DebugGameStudios** [comment]
If i right click on the executable and open properties my icons show on there ok. 

Just not the executable its self....really puzzling me.
![Screenshot (38)_LI](https://user-images.githubusercontent.com/56065423/73473664-044ba880-4385-11ea-8924-cbda7210d37b.jpg)

**Comment by KoBeWi** [comment]
^
Yeah, this is a cache problem mentioned by @pouleyKetchoupp. It should be gone when you move the exe to another directory. Or the icon is still Godot when you move file around?

Also the screenshot from Resource Hacker suggests that there really is something wrong with other icon sizes. Are you sure you set them up correctly?

**Comment by DebugGameStudios** [comment]
So after posting my last post i closed the properties and now it worked lol 

@KoBeWi its all sorted now i cleared all cache in the first place not sure why its lingering lol

Thank You all that helped me out.

Have you all been having same problems ? 

Its puzzling me if its the cache as i cleared it lol well at least its sorted now 👍   
 
![Screenshot (40)_LI](https://user-images.githubusercontent.com/56065423/73474189-f21e3a00-4385-11ea-93b8-4e9c96c26c4e.jpg)

**Comment by DebugGameStudios** [comment]
**Fixed....**

So just as a test i build another project and same thing was happening i don't believe its Godot.
It is the cache somehow....even tho i have cleared all cache  windows still giving me the Godot icons.

If i right click on the executable go to properties and then close the properties the icon then appears lol.

Once again Thank You all 👍

**Comment by bbacle** [comment]
I just was able to get it to work by changing the name of the folder where I was exporting my .exe file to.
And I was able to pack both files into one .exe file & the icon changed to the one I wanted.
NOTE: I have been working on this all afternoon, & it is now 5:00 pm for a total of 4 hours before I finally figured this out.
I had tried all of the other suggestions, but all I had to do was to change the name of the folder where I exported it to.
Hope that helps someone.

**Comment by Calinou** [comment]
> I just was able to get it to work by changing the name of the folder where I was exporting my .exe file to.

This is due to Windows caching executable icons :slightly_smiling_face:

**Comment by bbacle** [comment]
But where does Windows cache those icons so I can clear them out?

**Comment by Calinou** [comment]
> But where does Windows cache those icons so I can clear them out?

Using Windows' disk cleanup tool should do the job, but you may have to reboot to see the cache effectively being cleared.

It'll always be faster to export the executable with an unique name every time.

**Comment by bbacle** [comment]
Now that's a solution that I can sink my teeth into - change the name of the folder or use a unique name for the executable.
Both of those ideas should work.
Thanks.

**Comment by damzen** [comment]
> I just was able to get it to work by changing the name of the folder where I was exporting my .exe file to.
> And I was able to pack both files into one .exe file & the icon changed to the one I wanted.
> NOTE: I have been working on this all afternoon, & it is now 5:00 pm for a total of 4 hours before I finally figured this out.
> I had tried all of the other suggestions, but all I had to do was to change the name of the folder where I exported it to.
> Hope that helps someone.


My solution was to just rename the exported .exe in the Export settings in Godot. I believe windows is cacheing the last exported .exe file and icon. If you keep the ,.exported name of your .exe the same in Godot and change icon, it will use the last saved icon you exported. Very weird. Hope this helps others.

**Comment by SuzukaDev** [comment]
Im unable to get the icons working. I followed the documentation, tried everything, used different rcedit versions. Nothing works.

In case is useful:
In windows, if you want to refresh the icon cache:
Run (Windows+R): `ie4uinit.exe -show`

**Comment by blepis** [comment]
Not sure if this will work for you but I just got the same issue as you do. It fixed when I changed the folder name to be different from the .exe name.

**Comment by jackobear** [comment]
Also be sure to check "Modify Resources" in the export window just above where you pick the icon.  All icons and file metadata will be ignored otherwise.  Note that the docs never mention this flag.

**Comment by Calinou** [comment]
> Also be sure to check "Modify Resources" in the export window just above where you pick the icon. All icons and file metadata will be ignored otherwise. Note that the docs never mention this flag.

**Modify Resources** appears to be checked by default in 4.0.beta10, but feel free to open a pull request on https://github.com/godotengine/godot-docs/blob/master/tutorials/export/changing_application_icon_for_windows.rst either way :slightly_smiling_face:
